File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int ans = 0;
        unordered_set<int>cset(nums.begin(), nums.end());
        for(auto i : cset){
            if(cset.count(i-1)){
                continue;
            }
            int t = i;
            int len = 1;
            while(cset.count(++t)){
                len++;
            }
        ans = max(ans, len);
        }
       return ans; 
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seen = set(nums)
        longest = 0

        for num in seen:
            if num-1 not in seen:
                curr_num = num
                count = 1
                while curr_num + 1 in seen:
                    count += 1
                    curr_num += 1
                longest = max(longest, count)

        return longest